### PR TITLE
[REF-3592]Promote `rx.progress` from radix themes

### PR DESCRIPTION
**Pull request by ElijahAhianyo**
Promote `rx._x.progress` to `rx.progress`

### CODE REVIEW

**Review by picklelo — APPROVED** [review]
I think we can remove the progress low-level docs from reflex-web as well

File: reflex/experimental/__init__.py [hunk 1]
```diff
@@ -43,7 +42,6 @@ def toast(self):
     client_state=ClientStateVar.create,
     hooks=hooks,
     layout=layout,
-    progress=progress,
```
Comment (masenf) [inline reflex/experimental/__init__.py:46]: can we add this to the ExperimentalNamespace with a promotion warning in case anyone was already using it

Comment (ElijahAhianyo) [reply reflex/experimental/__init__.py:46]: should we have/specify a removal version for promoted experimental features/components just like we do for deprecated features?

Comment (masenf) [reply reflex/experimental/__init__.py:46]: it's more of a courtesy really. anything in `rx._x` can be removed or changed at any time

**Review by masenf — APPROVED** [review]

File: reflex/components/radix/primitives/__init__.pyi [hunk 2]
```diff
@@ -6,4 +6,3 @@
 from .accordion import accordion as accordion
 from .drawer import drawer as drawer
 from .form import form as form
-from .progress import progress as progress
```
Comment (masenf) [inline reflex/components/radix/primitives/__init__.pyi:9]: ENG-3756

**Pull request merged**